nilearn/_utils/niimg.py

```diff
@@ -123,7 +124,7 @@ def load_niimg(niimg, dtype=None):
     """
     from ..image import new_img_like  # avoid circular imports
 
-    if isinstance(niimg, str):
+    if isinstance(niimg, (str, os.PathLike)):
```

Comment:
maybe call `_stringify_path` on it first and then just check `isinstance(niimg, str)` -- this way it will work with any nibabel version